server/src/test/java/io/crate/integrationtests/MetadataTrackerITest.java

```diff
@@ -276,6 +277,86 @@ public void test_drop_publication_stops_the_replication_on_subscriber() throws E
+        createPublicationWithoutUser("pub2", false, List.of("t2"));
+
+        executeOnPublisher("CREATE TABLE t3 (id INT, p INT) PARTITIONED BY (p)");
+        executeOnPublisher("INSERT INTO t3 (id, p) VALUES (1, 1), (2, 2)");
+        createPublicationWithoutUser("pub3", false, List.of("t3"));
+
+        createSubscription("sub1", List.of("pub1", "pub2", "pub3"));
+
+        assertBusy(() -> assertThat(isTrackerActive(), is(true)));
+
+
+        executeOnPublisher("DROP PUBLICATION pub2"); // exclude publication with regular table
+        executeOnPublisher("DROP PUBLICATION pub3"); // exclude publication with partitioned table
+
+
+
+        // Dropping of the one one multiple publications doesn't stop tracking of the whole subscription.
+        assertBusy(() -> assertThat(isTrackerActive(), is(true)));
+
+        // Subscriber keeps receiving updates from non-dropped publication's regular table.
+        assertBusy(() -> {
+                executeOnPublisher("INSERT INTO t1 (id) VALUES (3)");
+                var res = executeOnSubscriber("SELECT id FROM t1 ORDER BY id");
+                assertThat(res.rowCount(), greaterThan(2L));
+            }
+        );
+
+        // Tables belonging to the dropped publications turned into regular writable tables.
+        assertBusy(
+            () -> {
+                long rowCount;
+                try {
+                    var response = executeOnSubscriber("INSERT INTO t2 (id) VALUES(4)");
+                    rowCount = response.rowCount();
+                } catch (OperationOnInaccessibleRelationException e) {
+                    throw new AssertionError(e.getMessage());
+                }
+                assertThat(rowCount, is(1L));
+            }
+        );
+        assertBusy(
+            () -> {
+                long rowCount;
+                try {
+                    var response = executeOnSubscriber("INSERT INTO t3 (id, p) VALUES(4, 4)");
+                    rowCount = response.rowCount();
+                } catch (OperationOnInaccessibleRelationException e) {
+                    throw new AssertionError(e.getMessage());
+                }
+                assertThat(rowCount, is(1L));
+            }
+        );
+    }
+
+    public void test_subscription_to_multiple_publications_stops_when_all_publications_are_dropped() throws Exception {
+        executeOnPublisher("CREATE TABLE t1 (id INT)");
+        createPublication("pub1", false, List.of("t1"));
+
+        executeOnPublisher("CREATE TABLE t2 (id INT)");
+        createPublicationWithoutUser("pub2", false, List.of("t2"));
+
+
+        createSubscription("sub1", List.of("pub1", "pub2"));
+
+        assertBusy(() -> assertThat(isTrackerActive(), is(true)));
+
+        executeOnPublisher("DROP PUBLICATION pub1");
+        executeOnPublisher("DROP PUBLICATION pub2");
+
+        // Dropping all publications of the subscription stops tracking.
+        assertBusy(() -> assertThat(isTrackerActive(), is(true)));
```

Comment:
```suggestion
        assertBusy(() -> assertThat(isTrackerActive(), is(false)));
```

Reply:
Good catch, thx - I wonder why it passes locally regardless of true/false - something is not good either in implementation or in the way I test it

Reply:
it will be first true and turn into false. so when wrapped inside assertBusy it will succeed in both cases ;)